src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -2552,101 +2545,35 @@ protected internal override Expression VisitRuntimeVariables(RuntimeVariablesExp
 
             private static IRuntimeVariables MergeRuntimeVariables(IRuntimeVariables first, IRuntimeVariables second, int[] indexes)
             {
-                return new MergedRuntimeVariables(first, second, indexes);
+                return new RuntimeOps.MergedRuntimeVariables(first, second, indexes);
             }
 
             protected internal override Expression VisitParameter(ParameterExpression node)
             {
-                LocalVariable var;
-                if (_variables.TryGetValue(node, out var))
+                IStrongBox box = GetBox(node);
+                if (box == null)
                 {
-                    return Expression.Convert(
-                        Expression.Field(
-                            var.LoadFromArray(
```

Comment:
This indirection seems very roundabout; we can resolve the storage location's `IStrongBox` here and return a `Constant` node for it to access the `Value` field on, similar to what the compiler's runtime quoter does.